x-pack/platform/plugins/shared/streams/server/routes/internal/streams/crud/route.ts

```diff
@@ -38,6 +39,10 @@ export const listStreamsRoute = createServerRoute({
 
     const streamNames = streams.filter(({ exists }) => exists).map(({ stream }) => stream.name);
 
+    const failureStorePrivileges = await scopedClusterClient.asCurrentUser.security.hasPrivileges({
```

Comment:
I think this is not necessary, `ESQL` will return only results for the indices the user has privileges
For the case of failure stores, if they user doesn't have access it will return
<img width="2640" height="728" alt="image" src="https://github.com/user-attachments/assets/71462088-ae2a-40ed-963f-50b77ca64776" />
So potentially we could handle this same as the case where fs doesn't exists, in that case the number of documents will be `0`, and that won't add up to the totalDocs and hence it won't play a role in the percentage calc. That would make the code also cleaner because we won't need `considerFailedQuality` boolean flag in `DataQualityColumn` and also we could remove the branches there